akd/src/append_only_zks.rs

```diff
@@ -243,6 +200,77 @@ impl AzksElementSet {
     }
 }
 
+/// Parallelism configuration for [Azks]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord)]
+pub struct AzksParallelismConfig {
+    /// Parallelization for node insertion.
+    pub insertion: AzksParallelismOption,
+    /// Parallelization for node preloading, during insertion and auditing.
+    pub preload: AzksParallelismOption,
+}
+
+impl AzksParallelismConfig {
+    /// The default fallback parallelism for parallel azks operations, used when
+    /// available parallelism cannot be determined automatically at runtime. Should be > 1
+    const DEFAULT_FALLBACK_PARALLELISM: u32 = 32;
+
+    /// Instantiate a parallelism config with no parallelism set for all fields.
+    pub fn none() -> Self {
+        Self {
+            insertion: AzksParallelismOption::None,
+            preload: AzksParallelismOption::None,
+        }
+    }
+}
+
+impl Default for AzksParallelismConfig {
+    fn default() -> Self {
+        Self {
+            insertion: AzksParallelismOption::Dynamic(Self::DEFAULT_FALLBACK_PARALLELISM),
+            preload: AzksParallelismOption::Dynamic(Self::DEFAULT_FALLBACK_PARALLELISM),
+        }
+    }
+}
+
+/// Parallelism setting for a given field in [AzksParallelismConfig].
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord)]
+pub enum AzksParallelismOption {
+    /// No parallelism.
+    None,
```

Comment:
Shall we stick with `Disabled` like we used to? Just to avoid reusing `None` given its use in `Option`.  No super strong feelings on my part here, so please feel free to choose what you prefer.